### PR TITLE
Use user namespace of ssh key and templates

### PR DESCRIPTION
**Pull request by guangbochen**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
currently, both ssh-keys and templates use `harvester-system` namespace if it is not specified by the user, however, this will introduce an RBAC issue of putting user-created resources into the system namespace.

**Solution:**
Use user namespace of ssh-key and templates, the namespace is required and will be passed from the front-end.

**Related Issue:**
https://github.com/harvester/harvester/issues/724

**Test plan:**

### CODE REVIEW

File: pkg/api/store/namespace.go [hunk 1]
```diff
@@ -6,13 +6,8 @@ import (
 
 type NamespaceStore struct {
```
Comment (gitlawr) [inline pkg/api/store/namespace.go]: Remove this

Comment (guangbochen) [reply pkg/api/store/namespace.go]: done

**Review by gitlawr — APPROVED** [review]
LGTM

**Comment by gitlawr** [comment]
Need a rebase

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**